src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -25,16 +29,45 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
     }
 
     @Override
-    public List<BookDto> findAll() {
-        return bookRepository.findAll().stream()
+    public List<BookDto> getAll() {
+        return bookRepository.findAll()
+                .stream()
                 .map(bookMapper::toDto)
                 .collect(Collectors.toList());
     }
 
     @Override
-    public BookDto getBookById(Long id) {
+    public BookDto getById(Long id) {
         Book bookToFind = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find book with id: " + id));
         return bookMapper.toDto(bookToFind);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        isPresent(id);
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        isPresent(id);
+        Book bookForUpdate = bookMapper.toModel(bookRequestDto);
+        bookForUpdate.setId(id);
+        return bookMapper.toDto(bookRepository.save(bookForUpdate));
+    }
+
+    @Override
+    public List<BookDto> search(BookSearchParameters searchParameters) {
+        Specification<Book> bookSpecification = bookSpecificationBuilder.build(searchParameters);
+        return bookRepository.findAll(bookSpecification).stream()
+                .map(bookMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    private void isPresent(Long id) {
```

Comment:
when you are naming your method isSmth... it suggests that method returns boolean value. For the methods like this, it's better to call validateSomethingExists or something like that.